DFC.ServiceTaxonomy.GraphSync/CosmosDb/CosmosDbSubgraphQuery.cs

```diff
@@ -86,28 +91,9 @@ public Query Query
             {
                 this.CheckIsValid();
 
-                var contentTypes = string.Join(",",
-                    NodeLabels.Where(nodeLabel => !nodeLabel.Equals("Resource", System.StringComparison.InvariantCultureIgnoreCase)).ToArray());
-
-                var parameters = new Dictionary<string, object?>
-                {
-                    {"maxLevel", MaxPathLength},
-                    {"idPropertyValue", IdPropertyValue},
-                    {"relationshipFilter", RelationshipFilter},
-                    {"ContentType", contentTypes }
-                };
-
-#pragma warning disable IDE0008 // Use explicit type
-#pragma warning disable CS8602 // Dereference of a possibly null reference.
-                var id = IdPropertyValue.ToString().Split('/')[IdPropertyValue.ToString().Split('/').Length - 1];
-#pragma warning restore CS8602 // Dereference of a possibly null reference.
-#pragma warning restore IDE0008 // Use explicit type
-
-#pragma warning disable CS8620 // Argument cannot be used for parameter due to differences in the nullability of reference types.
-                return new Query(
-                    @$"select * from c where c.id = '{id}'",
-                    parameters);
-#pragma warning restore CS8620 // Argument cannot be used for parameter due to differences in the nullability of reference types.
+                var contentType = NodeLabels.First(nodeLabel => !nodeLabel.Equals("Resource", StringComparison.InvariantCultureIgnoreCase)).ToLower();
+                var id = IdPropertyValue.ToString()?.Split('/').Last();
```

Comment:
I think we can use the getcontenttypeandid method?

Reply:
amended to use getcontenttypeandid()